core/base/src/TObject.cxx

```diff
@@ -751,6 +751,10 @@ void TObject::UseCurrentStyle()
 ///  Using the kWriteDelete option a previous key with the same name is
 ///  deleted only after the new object has been written. This option
 ///  is safer than kOverwrite but it is slower.
+///  NOTE: Neither kOverwrite nor kWriteDelete reduces the size of a TFile--
+///  the space is simply freed up to be overwritten; in the case of a TTree,
+///  only the metadata is replaced, effectively making the data invisible
+///  without deleting it.
```

Comment:
> effectively making the data invisible without deleting it.

Well not in the "normal" case, where you replace the TKey of the TTree object by a later incarnation of the TTree object.  For example you write a TKey for the TTree that has information about entry 0 through 100, then add more entries and the replace it with a new TKey for the TTree that now has information entry entry 0 through 200.

Reply:
Hm. I am not quite sure I follow. You mean that if I open a `TTree`, append some entries, then save it using `TObject::kOverwrite`, effectively only the new events are written? Whereas if I create a new `TTree`, then write it to a `TFile` in `UPDATE` mode that contains an old `TTree` with the same name, then write the new `TTree` using `TObject::kOverwrite`, the old `TTree` persists invisibly?

Reply:
> You mean that if I open a TTree, append some entries, then save it using TObject::kOverwrite, effectively only the new events are written?

Yes ... of course ... :)  

> the old TTree persists invisibly?

Well the data will .. the meta-data (the TTree itself) might or might not depending on the flag used (with kOverwrite then it will be gone) either way the original data will still be taking space in the file but will be hard to reach/retrieve.


With an ASCII picture.  File created with a TTree with one branch with 2 baskets. (also the output TFile::Map)

```
[file header][original tree basket 1][original tree basket 2][TTree object v1][TFile directory information][StreamerInfo information]
```

Then after opening the file, loading the TTree and appending a few entries, default mode with cycles.
```
[file header][original tree basket 1][original tree basket 2][TTree object v1][original tree basket 3][TTree object v2][[TFile directory information][StreamerInfo information]
```
Then after opening the file, loading the TTree and appending a few entries, default mode with overwrite 
```
[file header][original tree basket 1][original tree basket 2][TTree object v2][original tree basket 3][TFile directory information][StreamerInfo information]
```
or
```
[file header][original tree basket 1][original tree basket 2][gap][original tree basket 3][TFile directory information][TTree object v2][StreamerInfo information]
```
if the onfile size of v2 is larger than v1 (because there more basket information to store)

And now the same with the update the file but create a brand-new TTree with the same name (which essentially is assumed to never happens but it not actively prevented)

with cycles (original TTree still accessible via the older cycle]

```
[file header][original tree basket 1][original tree basket 2]  [**NEW TTree object**]  [original tree basket 3][TTree object v2][**new tree basket 1**][TFile directory information][StreamerInfo information]
```

with overwrite (original TTree meta data is gone from the disk)

```
[file header][original tree basket 1][original tree basket 2]  [**NEW TTree object**]  [original tree basket 3][**NEW tree basket 1**][TFile directory information][StreamerInfo information]
```

Cheers,
Philippe

Reply:
Ah, okay. I have updated the docs accordingly.